L1Trigger/Phase2L1GT/plugins/L1GTDeltaCut.h

```diff
@@ -0,0 +1,268 @@
+
+          double error = std::abs(precInvMass - lutInvMass);
+          massErrors.emplace_back(InvariantMassError{error, error / precInvMass, precInvMass});
+        }
+      }
+
+      if (minPTSquared_ || maxPTSquared_) {
+        int64_t pTSquared = obj1.hwPT().to_int64() * obj1.hwPT().to_int64() *
+                                static_cast<int64_t>(std::round(cosPhiLUT_.output_scale())) +
+                            obj2.hwPT().to_int64() * obj2.hwPT().to_int64() *
+                                static_cast<int64_t>(std::round(cosPhiLUT_.output_scale())) +
+                            2 * obj1.hwPT().to_int64() * obj2.hwPT().to_int64() * lutCosDPhi;
+        res &= minPTSquared_ ? pTSquared > std::round(minPTSquared_.value() * cosPhiLUT_.output_scale()) : true;
+        res &= maxPTSquared_ ? pTSquared < std::round(maxPTSquared_.value() * cosPhiLUT_.output_scale()) : true;
+      }
+
+      if (minTransMassSqrDiv2_ || maxTransMassSqrDiv2_) {
+        int64_t transMassDiv2 = obj1.hwPT().to_int64() * obj2.hwPT().to_int64() *
+                                (static_cast<int64_t>(coshEtaLUT_.output_scale()) - lutCosDPhi);
+        res &= minTransMassSqrDiv2_
+                   ? transMassDiv2 > std::round(minTransMassSqrDiv2_.value() * coshEtaLUT_.output_scale())
+                   : true;
+        res &= maxTransMassSqrDiv2_
+                   ? transMassDiv2 < std::round(maxTransMassSqrDiv2_.value() * coshEtaLUT_.output_scale())
+                   : true;
+      }
+
+      return res;
+    }
+
+    static void fillLUTDescriptions(edm::ParameterSetDescription& desc) {
+      edm::ParameterSetDescription coshLUTDesc;
+      L1GTSingleInOutLUT::fillLUTDescriptions(coshLUTDesc);
+      desc.add<edm::ParameterSetDescription>("cosh_eta_lut", coshLUTDesc);
+
+      edm::ParameterSetDescription coshLUT2Desc;
+      L1GTSingleInOutLUT::fillLUTDescriptions(coshLUT2Desc);
+      desc.add<edm::ParameterSetDescription>("cosh_eta_lut2", coshLUT2Desc);
+
+      edm::ParameterSetDescription cosLUTDesc;
+      L1GTSingleInOutLUT::fillLUTDescriptions(cosLUTDesc);
+      desc.add<edm::ParameterSetDescription>("cos_phi_lut", cosLUTDesc);
+    }
+
+    static void fillDescriptions(edm::ParameterSetDescription& desc) {
+      desc.addOptional<double>("minDEta");
+      desc.addOptional<double>("maxDEta");
+      desc.addOptional<double>("minDPhi");
+      desc.addOptional<double>("maxDPhi");
+      desc.addOptional<double>("minDR");
+      desc.addOptional<double>("maxDR");
+      desc.addOptional<double>("minDz");
+      desc.addOptional<double>("maxDz");
+      desc.addOptional<double>("minInvMass");
+      desc.addOptional<double>("maxInvMass");
+      desc.addOptional<double>("minTransMass");
+      desc.addOptional<double>("maxTransMass");
+      desc.addOptional<double>("minCombPt");
+      desc.addOptional<double>("maxCombPt");
+      desc.addOptional<bool>("os", false);
+      desc.addOptional<bool>("ss", false);
```

Comment:
Using these two as required parameters would avoid the need for `exists()` in the constructor
```suggestion
      desc.add<bool>("os", false);
      desc.add<bool>("ss", false);
```